File: Java/Swap_Nodes_in_Pairs.java
/**
 * Problem : https://leetcode.com/problems/swap-nodes-in-pairs/
 * 
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode swapPairs(ListNode head) {
        if(head == null) return head;
        ListNode root = new ListNode(0);
        ListNode next = root;
        do {
            ListNode node1 = head;
            ListNode node2 = head.next;
            if(node1 == null || node2 == null) {
                next.next = node1;
                break;
            }
            node1.next = node2.next;
            node2.next = node1;
            next.next = node2;
            head = head.next;
            next = next.next.next;
        }while(head != null);
        return root.next;
    }
}

File: Swift/swap-nodes-in-pairs.swift
/**
 * https://leetcode.com/problems/swap-nodes-in-pairs/
 * 
 * 
 */ 
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public var val: Int
 *     public var next: ListNode?
 *     public init(_ val: Int) {
 *         self.val = val
 *         self.next = nil
 *     }
 * }
 */
class Solution {
    func swapPairs(_ head: ListNode?) -> ListNode? {
        let root = ListNode(0)
        root.next = head
        var node = root
        while let l1 = node.next, let l2 = l1.next {
            l1.next = l2.next
            l2.next = l1
            node.next = l2
            node = l1
        }
        return root.next
    }
}
